nixos/modules/services/matrix/meowlnir.nix

```diff
@@ -0,0 +1,263 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.meowlnir;
+  format = pkgs.formats.yaml { };
+
+  settingsFile = "${cfg.dataDir}/config.yaml";
+  settingsFileUnsubstituted = format.generate "meowlnir-config-unsubstituted.yaml" cfg.settings;
+
+  registrationFile = "${cfg.dataDir}/registration.yaml";
+  registrationFileUnsubstituted = format.generate "meowlnir-registration-unsubstituted.yaml" cfg.registration;
+
+  mkDefaults = lib.mapAttrsRecursive (n: v: lib.mkDefault v);
```

Comment:
`mapAttrsRecursive` does not go into lists, right? So setting `logging.writers = [ ... ]` would discard the default here I think.

Reply:
The apply might save us here? I'm really not sure what the best approach is for this. https://github.com/NixOS/nixpkgs/pull/389114/files#diff-f4f9ef04020c1a34e25e3c790873c1b5fa8cf6576cbcad5f1de2d6230be3ae59R22